### PR TITLE
Backport aria-* from master

### PR DESCRIPTION
**Pull request by bugalot**
This is the backport of the [`aria-*` attributes feature](https://github.com/oat-sa/qti-sdk/pull/197) performed on `master` version. Please have a deep look at it to understand the scope of the PR. This should result in a release `0.21.0` of QTI-SDK `legacy`.

Compared to the `master` approach, the major obstacle here is that we did not have yet "QTI Version Management" inside `Marshaller` components was not included yet. I had to backport it within the scope of this `aria-*` feature. That is why we have so much changes.

**Please be aware that that such a PR cannot be merged without a FULL INTEGRATION TESTING in TAO Current Gen**. With the millions of Assessment Materials around the world, and all the Assessment Materials that our customers have to deal with. Lots of changes were performed. **Failure is not an option as it could have catastrophic impacts on all our customer base**. Another round of QA like https://oat-sa.atlassian.net/browse/QI-316 must be performed, in addition with specific `aria-*` contents ingestion/delivery to test takers to make sure everything is fine.

As a positive and motivational outcome, this is the road to a better integration of QTI 2.2 in TAO Current Gen. With QTI version support in QTI-SDK `legacy`, we can really tackle essential QTI 2.2 support like `dir`, `bidi`, `data-*` attributes. There is a big added value here !

**TAO Integration Testing branches ares the following. Please take extra care with reading carefully the instructions in these 2 Pull Requests.**

* https://github.com/oat-sa/lib-tao-qti/pull/72
* https://github.com/oat-sa/extension-tao-itemqti/pull/1483

### CODE REVIEW

File: qtism/data/storage/xml/XmlCompactDocument.php [hunk 1]
```diff
@@ -55,6 +55,24 @@
...
+     * Create a new XmlCompactDocument object.
+     *
+     * @param string $version
+     * @param QtiComponent|null $documentComponent
+     */
+    public function __construct($version = '2.1', QtiComponent $documentComponent = null)
+    {
+        // Version 1.0 was used in legacy code, let's keep it BC.
+        if ($version === '1.0') {
+            $version = '2.1.0';
```
Comment (gitromba) [inline qtism/data/storage/xml/XmlCompactDocument.php:70]: image

Comment (bugalot) [reply qtism/data/storage/xml/XmlCompactDocument.php:70]: Unfortunately required for legacy.

Comment (julien-sebire) [reply qtism/data/storage/xml/XmlCompactDocument.php:70]: This won't be enough to get everything aligned for legacy and master, but it will be part of another PR.

File: qtism/data/storage/xml/XmlCompactDocument.php [hunk 2]
```diff
@@ -380,4 +398,9 @@ public function explodeRubricBlocks()
 
         return $references;
     }
+
+    protected function inferVersion()
+    {
+        return '2.1';
```
Comment (julien-sebire) [inline qtism/data/storage/xml/XmlCompactDocument.php:404]: This will also need to be changed to enable compilation to QTI 2.2 tests. Will be part of another PR.
